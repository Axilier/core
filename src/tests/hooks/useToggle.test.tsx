import React from 'react';
import { mount } from 'enzyme';
import { useToggle } from '../../index';

// region demo
const Demo = (): JSX.Element => {
    const [state, toggleState] = useToggle(true);
    return (
        <div id={'container'}>
            <div id={'state'}>{state ? 1 : 0}</div>
            <button type={'button'} id={'toggle'} onClick={() => toggleState()}>
                toggle
            </button>
            <button
                type={'button'}
                id={'setFalse'}
                onClick={() => toggleState(false)}
            >
                setToFalse
            </button>
        </div>
    );
};
// endregion

const demo = mount(<Demo />);

describe('when I render the demo in an application', () => {
    it('should render correctly', () => {
        expect(demo).toMatchSnapshot();
    });
    it('should have an initial state of true', () => {
        expect(demo.find('#state').text()).toBe('1');
    });
    describe('when I click the setFalse button', () => {
        beforeAll(() => demo.find('#setFalse').simulate('click'));
        it('should set the state to false', () => {
            expect(demo.find('#state').text()).toBe('0');
        });
    });
    describe('when I click the toggle button', () => {
        beforeAll(() => demo.find('#toggle').simulate('click'));
        it('should set toggle the state to true', () => {
            expect(demo.find('#state').text()).toBe('1');
        });
    });
});
