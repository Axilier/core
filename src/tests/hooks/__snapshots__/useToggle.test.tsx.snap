// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when I render the demo in an application should render correctly 1`] = `
<Demo>
  <div
    id="container"
  >
    <div
      id="state"
    >
      1
    </div>
    <button
      id="toggle"
      onClick={[Function]}
      type="button"
    >
      toggle
    </button>
    <button
      id="setFalse"
      onClick={[Function]}
      type="button"
    >
      setToFalse
    </button>
  </div>
</Demo>
`;
